### DISCUSSION TITLE
ReferenceError: document is not defined

### DISCUSSION THREAD

**Discussion by simbolmina** [mantinedev/mantine#1505] [Q&A] [answered]
Hello,

I tried adding rich text editor to a vanilla nextJS project and got "ReferenceError: document is not defined" error. 

`import { useState } from "react";
import { RichTextEditor } from "@mantine/rte";

const initialValue =
  "<p>Your initial <b>html value</b> or an empty string to init editor without value</p>";

function Editor() {
  const [value, onChange] = useState(initialValue);
  return <RichTextEditor value={value} onChange={onChange} />;
}

export default Editor;`

this is simply what [doc](https://mantine.dev/others/rte/) has.

my installed dependencies are fallowing ones:

`  "dependencies": {
    "@mantine/core": "^4.2.6",
    "@mantine/hooks": "^4.2.6",
    "@mantine/next": "^4.2.6",
    "@mantine/rte": "^4.2.6",
    "next": "12.1.6",
    "react": "18.1.0",
    "react-dom": "18.1.0"`

Should I do something different to use this component?

**Comment by rtivital** [comment, marked as answer]
https://mantine.dev/others/rte/#usage-with-nextjs
